prac_05/emails.py

```diff
@@ -0,0 +1,40 @@
+"""
+CP1404 2020 - Practical 5
+Student Name: Amy Robinson
+Program - Emails
+"""
+
+
+def main():
+    """Generate a dictionary of names and emails"""
+    email_dict = {}
+
+    email = input("Email: ")
+    while email != "":
+        name = extract_name_from_email(email)
+        verify_name = input(f"Is your name {name}? (Y/n) ").lower()
+
+        if verify_name == 'y' or verify_name == "":
+            email_dict[email] = name
+
+        elif verify_name == 'n' or verify_name == 'no':
+            name = input("Name: ").title()
+            email_dict[email] = name
+
+        else:
+            print("Invalid verification")
```

Comment:
What follows? Is the name not entered? how does the user know what has happened? maybe add "name could not be entered" after invalid message, or loop it to ask for verification again.

Reply:
Interesting, really good point and recommendation. Thank you, I changed it.